Implements user authentication

Da1ryman opened the pull request:
Adds user authentication functionality, including registration and login. Integrates JWT for token-based authentication.
Adds authentication middlewares to protect routes. Includes configuration settings for JWT secret.

Da1ryman committed:
Implements user authentication

Adds user authentication functionality, including registration and login.
Integrates JWT for token-based authentication.
Adds authentication middlewares to protect routes.
Includes configuration settings for JWT secret.

XeniaVi commented on server/src/service/auth.service.ts:
```diff
@@ -0,0 +1,62 @@
+      const user = await userService.userByEmail(email);
+
+      if (user) {
+        if (user.password === password) {
```
а пароль приходит и хранится в зашифрованном виде?

Da1ryman committed:
Implements user password hashing

Enhances user security by implementing password hashing using bcrypt during signup.
This ensures that user passwords are not stored in plain text.

Updates authentication to compare hashed passwords using bcrypt.
